distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP.enso

```diff
@@ -407,66 +146,112 @@ type HTTP
              import Standard.Base.Network.HTTP.HTTP_Method.HTTP_Method
              import Standard.Base.Network.HTTP.Request.Request
 
              example_request =
-                 form = [Form.text_field "name" "John Doe"]
+                 form = Map.from_vector [["name", "John Doe"]]
                  req = Request.new HTTP_Method.Post "http://httpbin.org/post" . with_form form
                  http = HTTP.new (timeout = (Duration.new seconds=30))
                  http.request req
-    request : Request -> Response ! Request_Error
-    request self req =
+
+       > Example
+         Send a Get request with headers, and auto-parse the result
+         Note: This example will send a network request.
+
+             import Standard.Base.Network.HTTP.HTTP
+             import Standard.Base.Network.HTTP.HTTP_Method.HTTP_Method
+             import Standard.Base.Network.HTTP.Request.Request
+
+             example_request =
+                 req = Request.new HTTP_Method.Get "http://httpbin.org/get" try_auto_parse_response=True
+                 res = Examples.http_client.request req
+                 res.body
+    request : Request -> Boolean -> Response ! Request_Error
+    request self req try_auto_parse_response=False =
+        check_output_context ~action =
+            if Context.Output.is_enabled then action else
+                Error.throw (Forbidden_Operation.Error "HTTP POST/PUT/PATCH are forbidden as the Output context is disabled.")
         handle_request_error =
             Panic.catch JException handler=(cause-> Error.throw (Request_Error.Error 'IllegalArgumentException' cause.payload.getMessage))
-        Panic.recover Any <| handle_request_error <|
+
+        Panic.recover Any <| handle_request_error <| check_output_context <|
             body_publishers = HttpRequest.BodyPublishers
             builder = HttpRequest.newBuilder
+
             # set uri
-            builder.uri (Panic.rethrow req.uri.internal_uri)
-            # prepare headers and body
-            req_with_body = case req.body of
-                Request_Body.Empty ->
-                    Pair.new req body_publishers.noBody
-                Request_Body.Text text ->
-                    builder.header Header.text_plain.name Header.text_plain.value
-                    Pair.new req (body_publishers.ofString text)
-                Request_Body.Json json ->
+            uri = case req.uri of
+                _ : Text -> req.uri.to_uri
+                _ : URI -> req.uri
+            builder.uri uri.internal_uri
+
+            # Generate body publisher, possibly generating additional headers
+            body_publisher_and_headers = case req.body of
+                Request_Body.Text text encoding content_type ->
+                    header = Header.content_type content_type
+                    builder.header header.name header.value
+                    Pair.new (body_publishers.ofString text encoding.to_java_charset) []
```

Comment:
I think we should also add a `Content-Encoding` header.

Reply:
Done